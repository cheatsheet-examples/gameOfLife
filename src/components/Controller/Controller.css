.controls {
    margin-top: 20px;
}

.button {
    margin-left: 12px;
    cursor: pointer;
}