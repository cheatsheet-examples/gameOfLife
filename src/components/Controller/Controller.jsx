import React from 'react';
import './Controller.css';
import PropTypes from 'prop-types';

const Controller = ({ stopGame, playGame, generateRandomBoard, clearCells,
  isGameRunning }) => (
    <div className="controls">

      {isGameRunning ?
        <button className="button" onClick={stopGame}>Stop</button> :
        <button className="button" onClick={playGame}>Play</button>
      }
      <button className="button" onClick={generateRandomBoard}>Random</button>
      <button className="button" onClick={clearCells}>Clear</button>

    </div>
  );

Controller.propTypes = {
  stopGame: PropTypes.func.isRequired,
  playGame: PropTypes.func.isRequired,
  generateRandomBoard: PropTypes.func.isRequired,
  clearCells: PropTypes.func.isRequired,
  isGameRunning: PropTypes.bool.isRequired
};

Controller.defaultProps = {
  isGameRunning: false
};
export default Controller;