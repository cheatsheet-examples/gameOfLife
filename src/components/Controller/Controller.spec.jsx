import React from 'react';
import { shallow } from 'enzyme';
import Controller from './Controller';

const MOCK_STOP_GAME = jest.fn();
const MOCK_PLAY_GAME = jest.fn();
const MOCK_HANDLE_RANDOM = jest.fn();
const MOCK_HANDLE_CLEAR = jest.fn();
const MOCK_IS_GAME_RUN = false;

describe('Controller', () => {
  let wrapper;
  beforeEach(() => (wrapper = shallow(<Controller stopGame={MOCK_STOP_GAME} playGame={MOCK_PLAY_GAME}
    generateRandomBoard={MOCK_HANDLE_RANDOM} clearCells={MOCK_HANDLE_CLEAR} isGameRunning={MOCK_IS_GAME_RUN}
  />)));

  it('should render correctly', () => expect(wrapper).toMatchSnapshot());

  it('should render a <div />', () => {
    expect(wrapper.find('div').length).toEqual(1);
  });


});