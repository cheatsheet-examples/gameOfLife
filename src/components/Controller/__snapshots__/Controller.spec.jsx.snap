// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controller should render correctly 1`] = `
<div
  className="controls"
>
  <button
    className="button"
    onClick={[MockFunction]}
  >
    Play
  </button>
  <button
    className="button"
    onClick={[MockFunction]}
  >
    Random
  </button>
  <button
    className="button"
    onClick={[MockFunction]}
  >
    Clear
  </button>
</div>
`;
